packages/react-core/src/components/NumberInput/examples/WithStatus.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { NumberInput } from '@patternfly/react-core';
+
+export const WithStatus: React.FunctionComponent = () => {
+  const max = 10;
+  const min = 0;
+
+  const [validated, setValidated] = React.useState<'success' | 'error' | 'warning'>('success');
```

Comment:
I wonder if we should used an exported type here rather than expecting products to have to use this union explicitly in their typescript code.